### ISSUE TITLE
web: Use the path of the listening address

### ISSUE THREAD

**Issue by johanneslarsson** [open-policy-agent/opa#2906]
## Expected Behavior
We use path-based routing. Currently all OPA instances listen to the same path. Starting opa with a path in the listening address should prepend that to the mainRouter's context-path.

Start
``` 
opa % opa run -s blocked.tar.gz -a localhost:8181/blocked     
{"addrs":["localhost:8181/blocked"],"diagnostic-addrs":[],"level":"info","msg":"Initializing server.","time":"2020-11-13T15:33:44+01:00"}
```

Request
```
curl "http://localhost:8181/blocked/v1/data
{
  "result": {
    "devices": [
      "alice ipad"
    ],
    "example": {
      "blocklist": {
        "allow": true,
        "test_allow": true,
        "test_device_id_is_blocked": true,
        "test_user_id_is_blocked": true
      }
    },
    "users": [
      "bob"
    ]
  }
}
```

## Actual Behavior
Request
```
curl "http://localhost:8181/blocked/v1/data
404 page not found
```
## Steps to Reproduce the Problem
See above.

OPA version (edge)
```
opa version
Version: 0.25.0-dev
Build Commit: 3b16232
Build Timestamp: 2020-11-09T20:23:12Z
Build Hostname: fb36c9ac66ad
```

## Additional Info
The alternative solution is to introduce a new flag that for this prepended context path.

**Comment by tsandall** [comment]
It sounds like OPA is being deployed as service and then accessed across the network. Is that the case? Typically this kind of thing would be handled by a reverse-proxy like Nginx.

**Comment by johanneslarsson** [comment]
> It sounds like OPA is being deployed as service and then accessed across the network. Is that the case? Typically this kind of thing would be handled by a reverse-proxy like Nginx.

Correct, yes adding a reverse-proxy is a solution for this type of deployment. If this is a common deployment I could start on a PR to implement this in gorilla/mux instead. But if it isn't we can just close this issue and say use haproxy/nginx for path-based routing.

**Comment by tsandall** [comment]
Let's just leave this open for the time being. If we find it's heavily requested we can add it--however, for now, we'd recommend putting a reverse proxy in front.

**Comment by AdrianDiazCode** [comment]
I need this feature too :(

**Comment by FredrikBohman** [comment]
+1 for feature request

**Comment by LionOnTheChase** [comment]
This is an important feature. We have strict guardrails to only use URLs that are REST compliant. 

We would be forced to introduce a new container just to translate the urls. new container means extra cost.
